api/src/pcapi/core/offers/repository.py

```diff
@@ -591,29 +591,45 @@ def _filter_collective_offers_by_statuses(query: BaseQuery, statuses: list[str]
             allowed_expired_booking_status.add(educational_models.CollectiveBookingStatus.REIMBURSED)
 
+    if DisplayedStatus.EXPIRED.value in statuses:
         on_booking_status_filter.append(
             and_(
-                offer_id_with_booking_status_subquery.c.status.in_(allowed_expired_booking_status),
-                educational_models.CollectiveOffer.status == offer_mixin.CollectiveOfferStatus.EXPIRED.name,
+                educational_models.CollectiveOffer.hasBeginningDatetimePassed == True,
+                offer_id_with_booking_status_subquery.c.status == None,
+                educational_models.CollectiveOffer.validation == offer_mixin.OfferValidationStatus.APPROVED,
                 educational_models.CollectiveOffer.isArchived == False,
-            )
+            ),
         )
 
-    if DisplayedStatus.EXPIRED.value in statuses:
-        allowed_expired_booking_status.update(
-            {
-                educational_models.CollectiveBookingStatus.PENDING,
-                educational_models.CollectiveBookingStatus.CANCELLED,
-            }
+        on_booking_status_filter.append(
+            and_(
+                educational_models.CollectiveOffer.hasBeginningDatetimePassed == True,
+                offer_id_with_booking_status_subquery.c.status.in_(
+                    [educational_models.CollectiveBookingStatus.PENDING, None]
+                ),
+            ),
         )
 
+        allowed_expired_booking_status.update({None, educational_models.CollectiveBookingStatus.PENDING})
+        if not FeatureToggle.ENABLE_COLLECTIVE_NEW_STATUSES.is_active():
+            allowed_expired_booking_status.add(educational_models.CollectiveBookingStatus.CANCELLED)
+
+    if DisplayedStatus.CANCELLED.value in statuses and FeatureToggle.ENABLE_COLLECTIVE_NEW_STATUSES.is_active():
         on_booking_status_filter.append(
             and_(
-                or_(
-                    offer_id_with_booking_status_subquery.c.status == None,
-                    offer_id_with_booking_status_subquery.c.status.in_(allowed_expired_booking_status),
+                offer_id_with_booking_status_subquery.c.status.in_(
+                    [educational_models.CollectiveBookingStatus.CANCELLED]
                 ),
-                educational_models.CollectiveOffer.status == offer_mixin.CollectiveOfferStatus.EXPIRED.name,
+                educational_models.CollectiveOffer.validation == offer_mixin.OfferValidationStatus.APPROVED,
+                educational_models.CollectiveOffer.isArchived == False,
+            )
+        )
+
+    if len(allowed_expired_booking_status) > 0:
+        on_booking_status_filter.append(
+            and_(
+                offer_id_with_booking_status_subquery.c.status.in_(allowed_expired_booking_status),
+                educational_models.CollectiveOffer.hasBeginningDatetimePassed == True,
                 educational_models.CollectiveOffer.isArchived == False,
             )
         )
```

Comment:
Petit détail : je mettrais bien ce bloc avant le bloc `if DisplayedStatus.CANCELLED.value in statuses and FeatureToggle.ENABLE_COLLECTIVE_NEW_STATUSES.is_active():` pour qu'il soit plus près de la logique qui construit `allowed_expired_booking_status`